gprt/gprt.cpp

```diff
@@ -2316,38 +2377,115 @@ struct Context {
+      vkEnumerateDeviceExtensionProperties(device, nullptr, &extensionCount, nullptr);
+
+      std::vector<VkExtensionProperties> availableExtensions(extensionCount);
+      vkEnumerateDeviceExtensionProperties(device, nullptr, &extensionCount, availableExtensions.data());
+
+      std::set<std::string> requiredExtensions;
+      for (auto &cstr : deviceExtensions) {
+        requiredExtensions.insert(std::string(cstr));
+      }
+
+      for (const auto& extension : availableExtensions) {
+        requiredExtensions.erase(extension.extensionName);
+      }
+
+      return requiredExtensions.empty();
+    };
+
+    // Ray tracing related extensions required
+    enabledDeviceExtensions.push_back(VK_KHR_ACCELERATION_STRUCTURE_EXTENSION_NAME);
+    enabledDeviceExtensions.push_back(VK_KHR_RAY_TRACING_PIPELINE_EXTENSION_NAME);
+
+    // Required by VK_KHR_acceleration_structure
+    enabledDeviceExtensions.push_back(VK_KHR_BUFFER_DEVICE_ADDRESS_EXTENSION_NAME);
+    enabledDeviceExtensions.push_back(VK_KHR_DEFERRED_HOST_OPERATIONS_EXTENSION_NAME);
+    enabledDeviceExtensions.push_back(VK_EXT_DESCRIPTOR_INDEXING_EXTENSION_NAME);
+
+    enabledDeviceExtensions.push_back(VK_KHR_SHADER_NON_SEMANTIC_INFO_EXTENSION_NAME);
+
+    // Required for VK_KHR_ray_tracing_pipeline
+    enabledDeviceExtensions.push_back(VK_KHR_SPIRV_1_4_EXTENSION_NAME);
+
+    enabledDeviceExtensions.push_back(VK_KHR_RAY_QUERY_EXTENSION_NAME);
+
+    // Required by VK_KHR_spirv_1_4
+    enabledDeviceExtensions.push_back(VK_KHR_SHADER_FLOAT_CONTROLS_EXTENSION_NAME);
+
+    if (requestedFeatures.window)
+    {
+    	// If the device will be used for presenting to a display via a swapchain 
+      // we need to request the swapchain extension
+    	enabledDeviceExtensions.push_back(VK_KHR_SWAPCHAIN_EXTENSION_NAME);
+    }
+
+    #if defined(VK_USE_PLATFORM_MACOS_MVK) && (VK_HEADER_VERSION >= 216)
+      enabledDeviceExtensions.push_back(VK_KHR_PORTABILITY_SUBSET_EXTENSION_NAME);
+    #endif
+
     // Select physical device to be used
     // Defaults to the first device unless specified by command line
-    uint32_t selectedDevice = 0; // TODO
+    uint32_t selectedDevice = -1; // TODO
+
+    std::vector<uint32_t> usableDevices;
     
-    std::cout << "Available Vulkan devices" << "\n";
+    LOG("Searching for usable Vulkan physical device...");
     for (uint32_t i = 0; i < gpuCount; i++) {
       VkPhysicalDeviceProperties deviceProperties;
       vkGetPhysicalDeviceProperties(physicalDevices[i], &deviceProperties);
-      std::cout << "Device [" << i << "] : " << deviceProperties.deviceName << std::endl;
-      std::cout << " Type: " << physicalDeviceTypeString(deviceProperties.deviceType) << "\n";
-      std::cout << " API: " << (deviceProperties.apiVersion >> 22) << "."
-        << ((deviceProperties.apiVersion >> 12) & 0x3ff) << "."
-        << (deviceProperties.apiVersion & 0xfff) << "\n";
+      LOG("Device [" << i << "] : " << deviceProperties.deviceName);
```

Comment:
Why were these commented? Can we simply remove the lines?

Reply:
I added these back in, since it might be nice information still to have.